arch/x86/core/fatal.c

```diff
@@ -109,6 +109,25 @@ FUNC_NORETURN void _NanoFatalErrorHandler(unsigned int reason,
 	_SysFatalErrorHandler(reason, pEsf);
 }
 
+
+FUNC_NORETURN void _arch_syscall_oops(void *ssf_ptr)
+{
+	struct _x86_syscall_stack_frame *ssf =
+		(struct _x86_syscall_stack_frame *)ssf_ptr;
+	NANO_ESF oops_esf = {
+		.eip = ssf->eip,
+		.cs = ssf->cs,
+		.eflags = ssf->eflags
+	};
+
+	if (oops_esf.cs == USER_CODE_SEG) {
+		oops_esf.esp = ssf->esp;
```

Comment:
Doesn't this need to set the stack segment field too?

Reply:
NANO_ESF has no SS field.

Reply:
Also note, we have plans to cleanup and centralize _NanoFatalErrorHandler(). Currently it dumps out everything in NANO_ESF, even if some fields are unknown (it prints 0 or 0xdeadadead in these cases which is not ideal). in situations like this we ought to skip the register dump.